notebooks/ir_ima_visualization/ima_visualization_and_differencing.py

```diff
@@ -0,0 +1,327 @@
+
+        cube = cube[:,:,::-1]
+        integ_time = integ_time[::-1]
+    return cube, integ_time
+
+
+
+def compute_diff_imas(cube, integ_time, diff_method):
+    
+    '''
+    Compute the difference in signal between reads of a WFC3 IR IMA file.
+    
+    Parameters
+    ----------
+    cube : array-like
+       1024x1024xNSAMP datacube of the IR image in ascending time order, 
+       where NSAMP is the number of samples taken.
+
+    integ_time : array-like
+       Integration times associated with the datacube in ascending order.
+
+    diff_method: str
+       The method of finding the difference between reads. 
+       Either "instantaneous" or "cumulative".
+           
+    Returns
+    --------
+    diff : array-like
+        1024x1024x(NSAMP-1) datacube of the differebce between IR IMA reads in ascending time order, 
+        where NSAMP is the number of samples taken.
+        
+    '''
+    
+    if diff_method == 'instantaneous':
+        ima_j = cube[:, :, 1:]
+        ima_j_1 = cube[:,:,0:-1]
+        diff = ((ima_j*(integ_time[1:]-integ_time[0]))-\
+        (ima_j_1*(integ_time[0:-1]-integ_time[0])))/(integ_time[1:]-integ_time[0:-1])
+        
+    elif diff_method == 'cumulative':
+        diff = cube[:,:,0:-1] - cube[:,:,1:]
+        
+    else: # if an incorrect method is chosen raise an error
+        raise ValueError(f"{diff_method} is an invalid method. The allowed methods are 'instantaneous' and 'cumulative'.")
+
+        
+     
+    return diff
+
+
+def get_median_fullframe_lhs_rhs(cube, lhs_region, rhs_region):
+    
+    '''
+    Compute the median in the full-frame image, the user-defined left side region, and the user-defined right side region. 
+
+    Parameters
+    ----------
+    cube : array-like
+       1024x1024xNSAMP datacube of the IR image in ascending time order, 
+       where NSAMP is the number of samples taken.
+
+    lhs_region:  dict
+       The four corners (x0, x1, y0, y1) of the left hand region.
+
+    rhs_region : dict
+       The four corners (x0, x1, y0, y1) of the right hand region.
+           
+    Returns
+    -------
+    median_full_frame : array of floats
+        The median signal of the full frame of each read.
+        
+    median_lhs : array of floats
+        The median signal of the left side of each read.
+        
+    median_rhs : array of floats
+        The median signal of the right side of each read. 
+    '''
+    
+    
+    median_full_frame = np.nanmedian(cube, axis = (0,1))
+    median_lhs = np.nanmedian(cube[lhs_region['y0']:lhs_region['y1'],
+                                      lhs_region['x0']:lhs_region['x1'],:], axis = (0,1))
+    median_rhs = np.nanmedian(cube[rhs_region['y0']:rhs_region['y1'],
+                                      rhs_region['x0']:rhs_region['x1'],:], axis = (0,1))
+    
+    
+    return median_full_frame, median_lhs, median_rhs
+
+def get_median_std_fullframe_lhs_rhs(cube, lhs_region, rhs_region):
```

Comment:
Should be renamed `get_std_fullframe_lhs_rhs` since it's just computing standard deviations of cube, not of the medians. Documentation should be changed accordingly.